### PR TITLE
Renaming

### PR DESCRIPTION
**Pull request by wilwade**
Problem
=======
Names around batches and uri were confusing.
[#178801201](https://www.pivotaltracker.com/story/show/178801201)

Solution
========
Announce -> Publish
uri -> url

Change summary:
---------------
* Announce -> Publish
* uri -> url
* Updated packages
* Solc 0.8.4 -> 0.8.6
* Added indexes to Delegation events for searching consistency

### CODE REVIEW

**Review by enddynayn — APPROVED** [review]

**Comment by enddynayn** [comment]
fabulous!

File: hardhat.config.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ const getAccounts = (network: string) => {
 const config: HardhatUserConfig = {
   defaultNetwork: "hardhat",
   solidity: {
-    compilers: [{ version: "0.8.4", settings: {} }],
+    compilers: [{ version: "0.8.6", settings: {} }],
```
Comment (shannonwells) [inline hardhat.config.ts:28]: Do we intend to go out in September using the most up-to-date version of Solidity?

**Review by rosalinekarr — APPROVED** [review]

**Review by shannonwells — APPROVED** [review]
Looks ok

File: hardhat.config.ts (hunk 1 above)
Comment (wilwade) [reply hardhat.config.ts:28]: We will try

**Pull request merged**